apm-sniffer/apm-sdk-plugin/armeria-0.8.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/armeria/define/ArmeriaClientInstrumentation.java

```diff
@@ -0,0 +1,88 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.skywalking.apm.plugin.armeria.define;
+
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.ConstructorInterceptPoint;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.InstanceMethodsInterceptPoint;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.ClassInstanceMethodsEnhancePluginDefine;
+import org.apache.skywalking.apm.agent.core.plugin.match.ClassMatch;
+
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.takesArgument;
+import static net.bytebuddy.matcher.ElementMatchers.takesArguments;
+import static org.apache.skywalking.apm.agent.core.plugin.match.NameMatch.byName;
+
+/**
+ * @author kezhenxu94
+ */
+public class ArmeriaClientInstrumentation extends ClassInstanceMethodsEnhancePluginDefine {
+    private static final String ENHANCE_CLASS = "com.linecorp.armeria.client.UserClient";
+    private static final String INTERCEPTOR_CLASS = "org.apache.skywalking.apm.plugin.armeria.ArmeriaClientInterceptor";
+
+    @Override
+    protected ClassMatch enhanceClass() {
+        return byName(ENHANCE_CLASS);
+    }
+
+    @Override
+    public ConstructorInterceptPoint[] getConstructorsInterceptPoints() {
+        return new ConstructorInterceptPoint[0];
+    }
+
+    @Override
+    public InstanceMethodsInterceptPoint[] getInstanceMethodsInterceptPoints() {
+        return new InstanceMethodsInterceptPoint[]{
+            new InstanceMethodsInterceptPoint() {
+                @Override
+                public ElementMatcher<MethodDescription> getMethodsMatcher() {
+                    return named("execute")
+                        .and(takesArguments(7))
+                        .and(takesArgument(0, named("io.netty.channel.EventLoop")))
+                        .and(takesArgument(1, named("com.linecorp.armeria.common.HttpMethod")))
+                        .and(takesArgument(2, named("java.lang.String")))
+                        .and(takesArgument(3, named("java.lang.String")))
+                        .and(takesArgument(4, named("java.lang.String")));
```

Comment:
@trustin If you have time, could you take a look at here. This is a PR for Armenia SkyWalking plugin. Is this method a suitable injecting point to add SkyWalking codes for tracing Ameria RPC.

Reply:
@ikhoon I think you can also help on reviewing on this please?

Reply:
@kezhenxu94 Sure, why not. I'm happy to review this PR. 🥰